openshift-with-appstudio-test/e2e/jvm_build_service_test.go

```diff
@@ -0,0 +1,363 @@
+			}
+			podLog := string(b)
+			ta.t.Logf("pod logs for container %s in pod %s:  %s", container.Name, pod.Name, podLog)
+
+		}
+
+	}
+}
+
+func debugAndFailTest(ta *testArgs, failMsg string) {
+	dumpPods(ta)
+	ta.t.Fatalf(failMsg)
+}
+
+func setup(t *testing.T, ta *testArgs) *testArgs {
+	if ta == nil {
+		ta = &testArgs{
+			t:        t,
+			timeout:  time.Minute * 10,
+			interval: time.Minute,
+		}
+	}
+	setupClients(ta.t)
+
+	if len(ta.ns) == 0 {
+		ta.ns = generateName(testNamespace)
+		_, err := projectClient.ProjectV1().ProjectRequests().Create(context.Background(), &projectv1.ProjectRequest{
+			ObjectMeta: metav1.ObjectMeta{Name: ta.ns},
+		}, metav1.CreateOptions{})
+
+		if err != nil {
+			debugAndFailTest(ta, fmt.Sprintf("%#v", err))
+		}
+	}
+	ta.gitClone = &v1beta1.Task{}
+	obj := streamRemoteYamlToTektonObj("https://raw.githubusercontent.com/redhat-appstudio/build-definitions/main/tasks/git-clone.yaml", ta.gitClone, ta)
+	var ok bool
+	ta.gitClone, ok = obj.(*v1beta1.Task)
+	if !ok {
+		debugAndFailTest(ta, fmt.Sprintf("https://raw.githubusercontent.com/redhat-appstudio/build-definitions/main/tasks/git-clone.yaml did not produce a task: %#v", obj))
+	}
+	var err error
+	ta.gitClone, err = tektonClient.TektonV1beta1().Tasks(ta.ns).Create(context.TODO(), ta.gitClone, metav1.CreateOptions{})
+	if err != nil {
+		debugAndFailTest(ta, err.Error())
+	}
+
+	return ta
+}
+
+func projectCleanup(ta *testArgs) {
+	projectClient.ProjectV1().Projects().Delete(context.Background(), ta.ns, metav1.DeleteOptions{})
+}
+
+func decodeBytesToTektonObjbytes(bytes []byte, obj runtime.Object, ta *testArgs) runtime.Object {
+	decodingScheme := runtime.NewScheme()
+	utilruntime.Must(v1beta1.AddToScheme(decodingScheme))
+	decoderCodecFactory := serializer.NewCodecFactory(decodingScheme)
+	decoder := decoderCodecFactory.UniversalDecoder(v1beta1.SchemeGroupVersion)
+	err := runtime.DecodeInto(decoder, bytes, obj)
+	if err != nil {
+		debugAndFailTest(ta, err.Error())
+	}
+	return obj
+}
+
+func streamRemoteYamlToTektonObj(url string, obj runtime.Object, ta *testArgs) runtime.Object {
+	resp, err := http.Get(url)
+	if err != nil {
+		debugAndFailTest(ta, err.Error())
+	}
+	defer resp.Body.Close()
+	bytes, err := io.ReadAll(resp.Body)
+	if err != nil {
+		debugAndFailTest(ta, err.Error())
+	}
+	return decodeBytesToTektonObjbytes(bytes, obj, ta)
+}
+
+func streamFileYamlToTektonObj(path string, obj runtime.Object, ta *testArgs) runtime.Object {
+	bytes, err := ioutil.ReadFile(path)
+	if err != nil {
+		debugAndFailTest(ta, err.Error())
+	}
+	return decodeBytesToTektonObjbytes(bytes, obj, ta)
+}
+
+func TestExampleRun(t *testing.T) {
+	ta := setup(t, nil)
+	defer projectCleanup(ta)
+
+	path, err := os.Getwd()
+	if err != nil {
+		debugAndFailTest(ta, err.Error())
+	}
+	ta.t.Logf("current working dir: %s", path)
+
+	mavenYamlPath := filepath.Join(path, "..", "..", "deploy", "base", "maven-v0.2.yaml")
```

Comment:
Not a blocker, but should we eventually use the tasks from the build definitions bundle here rather than this one?

Reply:
absolutely

that said, as you noted recently in https://docs.google.com/document/d/1qJvVaLWiWtkC4Gd2Q6ehSpxHBQLjLDxoGPdY9f_oYcY/edit?disco=AAAAcWLy7Ec we are not quite there yet, right?

Although I have not tried it super recently, that has been my experience as well so far when I try java-builder component builds with the currently hacbs recipes in build-definitions.

but as soon as that changes, this bit should pivot to use the build-definition stuff